qiskit/optimization/converters/linear_equality_to_penalty.py

```diff
@@ -51,6 +60,17 @@ def encode(self, op: QuadraticProgram, penalty_factor: float = 1e5,
         self._src = copy.deepcopy(op)  # deep copy
         self._dst = QuadraticProgram()
 
+        # If penalty is None, set the penalty coefficient by _auto_define_penalty()
+        if self._penalty is None:
+            self._penalty = self._auto_define_penalty()
+        penalty = self._penalty
```

Comment:
What is the expectation ove re-use of this object. Should I be able to create an instance and re-use it convert (encode) a number of QuadraticPrograms. In the logic above self._penalty is overridden by the autocomputed value. So if penalty was passed in constructor of None and stored in self._penalty (as None) this will be lost and the 2nd (and any subsequent) conversions via encode will end up using the autocomputed penalty of the first conversion that was carried out. Maybe
`penalty = self._penalty if self._penalty is not None else self._auto_define_penalty()` or something like this so as to preserve the original penalty from the constructor.

And as aside, if this is intended for re-use does it make sense to have a self._dst and a self._src. I see them in other converters so maybe this is consistent with elsewhere. But src and dest are are really only used in encode which means they could entirely be local variables in that function to save holding references to src and dst objects from the last conversion

Reply:
Hi @woodsp-ibm, I've heard from @stefan-woerner that we do not re-use a converter object (more precisely we do not change the penalty factor once it's set) that's why self._penalty is overridden

Reply:
Hi, if people choose to use these convertors themselves then they can end up with unexpected behavior with this self._penalty override. In this case it seems less than obvious to me why it needs to be overridden and not just computed into a local variable leaving the original None.

If we truly do not expect these objects to be created and then be potentially used to do multiple encodings, where they may or may not work as expected based on values being giving in constructor being overridden, then at a bare minimum we should make this clear - maybe a Note: in the docstring in the init.py in this module. Since documentation is often not read, I would have thought it better to allow them to be used more than once since that seems more intuitive I think and what one might expect. Is making them behave like this a challenge? - for this one it seems simple. Arguably one could have convertors as simple methods, but in this case it seems a convertor is parameterized and then used later which I guess precludes this.